src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) CoSMo Software Consulting Pte. Ltd. - All Rights Reserved
+ */
+
+package io.cosmosoftware.kite.instrumentation;
+
+import io.cosmosoftware.kite.exception.KiteTestException;
+import io.cosmosoftware.kite.report.Status;
+
+import javax.json.JsonObject;
+
+public class NetworkProfile {
+
+  private final Profile profile;
+
+  private final String name;
+
+
+  /**
+   * Instantiates a new Instance.
+   *
+   * @param jsonObject the json object
+   */
+  public NetworkProfile(JsonObject jsonObject) throws KiteTestException {
+    String missingKey = "";
+    try {
+      missingKey = "name";
+      this.name = jsonObject.getString("name");
```

Comment:
if using missingKey then might as well use `this.name = jsonObject.getString(missingKey);`, or just call it key

Reply:
the correct way should be to implement our own
getJsonString(jsonObject, "name");

```java
public static String getJsonString(JsonObject jsonObject, String key) throws KiteTestException {
  try {
    return jsonObject.getString("name");
  } catch (NullPointerException e) {
    throw new KiteTestException("Error in json config instrumentation, the key " + key+ " is missing.", Status.BROKEN, e);
  }
}
```